drivers/sensor/tmp108/tmp108.c

```diff
@@ -0,0 +1,406 @@
+/*
+ * Copyright (c) 2021 Jimmy Johnson <catch22@fastmail.net>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_tmp108
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+#include <sys/util.h>
+#include <sys/byteorder.h>
+#include <logging/log.h>
+#include <kernel.h>
+
+#include "tmp108.h"
+
+/** TI conversion scale from 16 bit int temp value to float */
+
+#define TMP108_TEMP_MULTIPLIER	  62500
+
+/** TMP typical conversion time of 27 ms after waking from sleep */
+
+#define TMP108_WAKEUP_TIME_IN_MS 30
+
+#define LOG_LEVEL CONFIG_SENSOR_LOG_LEVEL
+
+LOG_MODULE_REGISTER(TMP108, CONFIG_SENSOR_LOG_LEVEL);
+
+struct tmp108_config {
+	const struct i2c_dt_spec i2c_spec;
+	const struct gpio_dt_spec alert_gpio;
+};
+
+int tmp108_reg_read(const struct device *dev, uint8_t reg, uint16_t *val)
+{
+	const struct tmp108_config *cfg = dev->config;
+	int result = 0;
+
+	result = i2c_burst_read_dt(&cfg->i2c_spec, reg, (uint8_t *) val, 2);
+
+	if (result < 0) {
+		return result;
+	}
+
+	*val = sys_be16_to_cpu(*val);
+
+	return 0;
+}
+
+int tmp108_reg_write(const struct device *dev, uint8_t reg, uint16_t val)
+{
+	const struct tmp108_config *cfg = dev->config;
+
+	uint8_t tx_buf[3] = { reg, val >> 8, val & 0xFF };
+
+	int result = i2c_write_dt(&cfg->i2c_spec, tx_buf, sizeof(tx_buf));
+
+	if (result < 0) {
+		return result;
+	}
+
+	return 0;
+}
+
+int tmp108_write_config(const struct device *dev, uint16_t mask, uint16_t conf)
+{
+	uint16_t config = 0;
+	int result = 0;
+
+	result = tmp108_reg_read(dev, TI_TMP108_REG_CONF, &config);
+
+	if (result < 0) {
+		return result;
+	}
+
+	config &= mask;
+	config |= conf;
+
+	result = tmp108_reg_write(dev, TI_TMP108_REG_CONF, config);
+
+	if (result < 0) {
+		return result;
+	}
+
+	return 0;
+}
+
+int ti_tmp108_read_temp(const struct device *dev)
+{
+	struct tmp108_data *drv_data = dev->data;
+	int result = 0;
+
+	/* clear previous temperature readings */
+
+	drv_data->sample = 0U;
+
+	/* Get the most recent temperature measurement */
+
+	result = tmp108_reg_read(dev, TI_TMP108_REG_TEMP, &drv_data->sample);
+
+	if (result < 0) {
+		return result;
+	}
+
+	return 0;
+}
+
+static int tmp108_sample_fetch(const struct device *dev,
+							enum sensor_channel chan)
```

Comment:
please, validate supported channels here (SENSOR_CHAN_ALL and SENSOR_CHAN_AMBIENT_TEMP)

Reply:
I'm still not sure when sensor channel all should be used, can you point me to some documentation?

Reply:
see for example https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/lm77/lm77.c#L272

ref. https://github.com/zephyrproject-rtos/zephyr/blob/main/include/drivers/sensor.h#L513